plugin-server/src/worker/vm/promise-manager.ts

```diff
@@ -23,12 +26,20 @@ export class PromiseManager {
         promise.finally(() => {
             this.pendingPromises.delete(promise)
         })
+
+        this.statsd?.increment('promise_manager_promises_tracked', {
+            instanceId: this.instanceId,
```

Comment:
Are these high cardinality tags? If so I this could cause issues with InfluxDB I think although not completely sure: https://docs.influxdata.com/influxdb/cloud/write-data/best-practices/resolve-high-cardinality/#learn-the-causes-of-high-series-cardinality

Anyway, not totally sure how it all works tbh but thought I'd mention in case we have slowdown in Grafana and might wonder why.

Reply:
I guess there'll only be a handful of threadIds so I guess should be fine 🤔

Reply:
they kind of are - I was a bit wary of doing this but we do need to filter by them in order to get meaningful insights.

Thread id has 5 possible options (MAIN, 1, 2, 3, 4) and instanceId changes with each deploy so reasonably high cardinality. I'll at least gather some data now and will revert if it messes things up

Reply:
Yeah sounds fine. As long as it's not like, every promise is a different unique threadId!